pvlib/tests/test_snow.py

```diff
@@ -95,3 +95,36 @@ def test_dc_loss_nrel():
     expected = pd.Series([1, 1, .5, .625, .25, .5, 0])
     actual = snow.dc_loss_nrel(snow_coverage, num_strings)
     assert_series_equal(expected, actual)
+
+
+def test__townsend_effective_snow():
+    snow_total = np.array([25.4, 25.4, 12.7, 2.54, 0, 0, 0, 0, 0, 0, 12.7,
+                           25.4])
+    snow_events = np.array([2, 2, 1, 0, 0, 0, 0, 0, 0, 0, 2, 3])
+    expected = np.array([19.05, 19.05, 12.7, 0, 0, 0, 0, 0, 0, 0, 9.525,
+                         254 / 15])
+    actual = snow._townsend_effective_snow(snow_total, snow_events)
+    np.testing.assert_allclose(expected, actual, rtol=1e-07)
+
+
+def test_loss_townsend():
+    snow_total = np.array([25.4, 25.4, 12.7, 2.54, 0, 0, 0, 0, 0, 0, 12.7,
+                           25.4])
+    snow_events = np.array([2, 2, 1, 0, 0, 0, 0, 0, 0, 0, 2, 3])
+    surface_tilt = 20
+    relative_humidity = np.array([80, 80, 80, 80, 80, 80, 80, 80, 80, 80,
+                                  80, 80])
+    temp_air = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+    poa_global = np.array([350000, 350000, 350000, 350000, 350000, 350000,
+                           350000, 350000, 350000, 350000, 350000, 350000])
+    angle_of_repose = 40
+    slant_height = 2.54
+    lower_edge_height = 0.254
+    expected = np.array([0.0007696253, 0.0007992262, 0.0006216201,
+                         0.0001715392, 0, 0, 0, 0, 0, 0, 0.0002643821,
+                         0.0006068194])
```

Comment:
These expected losses seem a couple orders of magnitude too small for the inputs, so I suspect this implementation is not quite correct yet.  I'm not immediately seeing any remaining input unit issues, but I do notice that the y-axis in Fig 7 is fractional (not percentage) but has the same slope/C1 value, so maybe Eqn 3 is mislabeled in the reference and actually calculates fractional loss, not percentage loss?  If that's true, these values ought to be 100x what they are now, which would be a lot more reasonable.

Reply:
loss_townsend now takes its Eqn 3 calculation as a fractional loss, and no longer divides it by 100. I also added a comment to explain why we think Eqn 3 calculates a fractional loss.